sa/database.go

```diff
@@ -76,6 +68,50 @@ func NewDbMap(driver string, dbConnect string) (*gorp.DbMap, error) {
 	return dbmap, err
 }
 
+// recombineURLForDB transforms a database URL to a URL-like string
+// that the mysql driver can use. The mysql driver needs the Host data
+// to be wrapped in "tcp()" but url.Parse will escape the parentheses
+// and the mysql driver doesn't understand them. So, we can't have
+// "tcp()" in the configs, but can't leave it out before passing it to
+// the mysql driver. Similarly, the driver needs the password and
+// username unescaped. Compromise by doing the leg work if the config
+// says the database URL's scheme is a fake one called
+// "mysqltcp://". See
```

Comment:
Stepping back a level: The goal of this function is to make sure no-one runs boulder without the `parseTime` param (and in future the strict mode param). With the current implementation, we only protect people if they include the `mysqltcp://` scheme.

I know you've put a bunch of work into getting this right, but what if we simplify it to just grep for `parseTime=true` anywhere in the string and error out, rather than trying to add it? That would let us protect people who don't think to include the `mysqltcp://` prefix.